siriuspy/siriuspy/pwrsupply/siggen.py

```diff
@@ -0,0 +1,159 @@
+            else self.aux_param[2]
+        self.aux_param[0] = float(theta_begin) if theta_begin is not None \
+            else self.aux_param[0]
+        self.aux_param[1] = float(theta_end) if theta_end is not None \
+            else self.aux_param[1]
+        self.aux_param[2] = float(decay_time) if decay_time is not None \
+            else self.aux_param[2]
+
+    # --- public methods ---
+
+    @property
+    def duration(self):
+        """Duration of signal [s]."""
+        return self.nr_cycles / self.freq
+
+    @property
+    def rampup_time(self):
+        """Rampup time for Trapezoidal signals."""
+        return self.aux_param[0]
+
+    @rampup_time.setter
+    def rampup_time(self, value):
+        """Set Rampup time for Trapezoidal signals."""
+        self.aux_param[0] = value
+        return value
+
+    @property
+    def theta_begin(self):
+        """Initial phase for Sine or DampedSine signals."""
+        return self.aux_param[0]
+
+    @theta_begin.setter
+    def theta_begin(self, value):
+        """Set Initial phase for Sine or DampedSine signals."""
+        self.aux_param[0] = value
+        return value
+
+    @property
+    def rampdown_time(self):
+        """Rampdown time for Trapezoidal signals."""
+        return self.aux_param[1]
+
+    @rampdown_time.setter
+    def rampdown_time(self, value):
+        """Set Rampdown time for Trapezoidal signals."""
+        self.aux_param[1] = value
+        return value
+
+    @property
+    def theta_end(self):
+        """Final phase for Sine or DampedSine signals."""
+        return self.aux_param[1]
+
+    @theta_end.setter
+    def theta_end(self, value):
+        """Set Final phase for Sine or DampedSine signals."""
+        self.aux_param[1] = value
+        return value
+
+    @property
+    def plateau_time(self):
+        """Plateau time for Trapezoidal signals."""
+        return self.aux_param[2]
+
+    @plateau_time.setter
+    def plateau_time(self, value):
+        """Set Plateau time for Trapezoidal signals."""
+        self.aux_param[2] = value
+        return value
+
+    @property
+    def decay_time(self):
+        """Decay time constant for DampedSine signals."""
+        return self.aux_param[2]
+
+    @decay_time.setter
+    def decay_time(self, value):
+        """Set Decay time constant for DampedSine signals."""
+        self.aux_param[2] = value
+        return value
+
+    def get_waveform(self, nr_points=100):
+        """Return list with signal waveform."""
+        t = _np.linspace(0.0, self.duration, nr_points)
+        if self.type in ('Sine', 'DampedSine'):
+            # TODO: confirm!
+            if self.type == 'Sine':
+                amp = self.amplitude * _np.ones(t.shape)
+            else:
+                amp = self.amplitude * _np.exp(-t/self.decay_time)
+            wfm = self.offset * _np.ones(t.shape)
+            phase = self.freq*t % (2*_np.pi)
```

Comment:
I think there is a bug here... the interpretation of `self.freq` is the frequency, not the angular frequency.

I think the correct code would be:

`phase = (2 * _np.pi) * (self.freq * t % 1)`

Reply:
you're right, @fernandohds564 . thanks!